File: python/resistor-color-duo/resistor_color_duo.py
RES_COLORS = [
            'black',
            'brown',
            'red',
            'orange',
            'yellow',
            'green',
            'blue',
            'violet',
            'grey',
            'white'
        ]


def value(colors):
    return int(''.join(str(RES_COLORS.index(color)) for color in colors[:2]))

File: ruby/resistor-color-duo/resistor_color_duo.rb
#!/usr/bin/env ruby -w

class ResistorColorDuo

	@@RESISTOR_COLORS = %w[
		black brown red orange yellow green blue violet grey white
		].freeze

	def self.value(args=[])
		args
		.take(2)
		.map {|color| @@RESISTOR_COLORS
		.index(color)}
		.join
		.to_i
	end

end
